fhir-persistence-schema/docs/DB2SchemaMigration.md

```diff
@@ -0,0 +1,382 @@
+
+For each of the above changes, the version of the table must be incremented. 
+
+The resource table stores the FHIR resource as a compressed blob, changes to the specification or extensions should only impact Search values. 
+
+The tables have various indices - PrimaryKey, Index and UniqueIndex. These indices are created as part of the Java object - Table.  These 
+
+If a modification is made to a Search Parameter (addition, removal), then the resource uses the existing search parameter values until a new version of the resource is processed.
+
+Each of these tables has row-level permissions based on the conditional READ-only global variable, such as: 
+
+``` sql
+CREATE PERMISSION FHIRAPP.AUDITEVENT_COMPOSITES_TENANT
+    ON FHIRAPP.AUDITEVENT_COMPOSITES FOR ROWS
+    WHERE FHIRAPP.AUDITEVENT_COMPOSITES.MT_ID = FHIR_ADMIN.SV_TENANT_ID 
+    ENFORCED FOR ALL ACCESS ENABLE ;
+```
+
+These permissions should not be removed, migrated, altered or changed.
+
+## Managing Stored Procedures
+
+`FHIRDATA.ADD_CODE_SYSTEM` is created in the default schema.  The stored procedures `add_code_system`, `add_parameter_name`, `add_resource_type`, and `add_any_resource`.
+
+If updates are made to the `add_code_system.sql`, the `FhirSchemaGenerator.buildSchema`'s FhirSchemaConstants.INITIAL_VERSION must be updated to the next highest integer (in this case 2). 
+
+``` java
+ProcedureDef pd = model.addProcedure(this.schemaName, 
+    ADD_CODE_SYSTEM, 
+    FhirSchemaConstants.INITIAL_VERSION, 
+    () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_CODE_SYSTEM.toLowerCase() + ".sql", null), 
+    Arrays.asList(fhirSequence, codeSystemsTable, allTablesComplete), 
+    procedurePrivileges);
+```
+
+When the `fhir-persistence-schema` buildSchema is executed, the **INSERT**, **SELECT**, **UPDATE**, **DELETE** grant is applied again, and the procedures are updated. Each Procedure privilege is reset upon re-executing the action. 
+
+If you change the stored procedure signature, you MUST drop the stored procedure, before applying the updated stored procedure to the database. 
+
+## Managing GRANTS
+The Db2 data definition secures data access using `GRANT` predicates, and updated and changed using the `--grant-to` predicate to apply the grants. 
+
+If a grant is removed from the Java code, a manual process must be followed to remove or change the grant for the corresponding tables, procedures and variables. 
+
+## References
+The following are useful references for inspecting and understanding the Db2 data definition, and checking the current configurations, and useful links. 
+
+### References: SQL Queries
+Included are useful queries to debug the Schema. 
+
+#### Tablespace: Check the EXTENTSIZE for a tablespace
+**Query**
+``` sql
+SELECT EXTENTSIZE FROM syscat.tablespaces WHERE tbspace = 'FHIR_TS'
+```
+**Result**
+``` sql
+EXTENTSIZE
+-----------
+          4
+  1 record(s) selected.
+```
+
```

Comment:
Maybe remove this one?  I'm really not sure when you'd want to do it and it doesn't seem germane to managing schema migrations.

Reply:
it is helpful to check the exntsize, I've removed per your request